org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Replay/ImportDynamicPolygon.java

```diff
@@ -139,7 +140,7 @@ public final Object readThisLine(final String theLine) throws ParseException
 		// start with the symbology
 		symbology = st.nextToken();
 
-		String theName = checkForQuotedName(st).trim();
+		//String theName = checkForQuotedName(st).trim();

```

Comment:
Yes, we need to use it, so that the Polygon gets put into the correct group.

All of these polygons are meant to go into the same folder:
![image](https://user-images.githubusercontent.com/1108513/47148696-49031b80-d2c9-11e8-919a-96da1a6f1e1e.png)

But, since we fail to give the shape a name, it goes into default annotations level.

This is where the polygon should go:
![image](https://user-images.githubusercontent.com/1108513/47149090-7ef4cf80-d2ca-11e8-866d-b3602c267d38.png)

To fix this, you could make a call to
`
sw.setTrackName(theName);
`

But, instead, removing `setTrackName()` from DynamicPolygonShapeWrapper, including the name in the constructor, and making it a final member field.

You'll also have to make another change. `ImportReplay()` will have to be modified at line 2240 to check for `IDynamicShapeWrapper` rather than `DynamicShapeWrapper`.